realm/src/main/java/io/realm/internal/RealmCore.java

```diff
@@ -39,7 +38,7 @@
     private static final String JAVA_LIBRARY_PATH = "java.library.path";
 //*/
 
-    private static AtomicBoolean libraryIsLoaded = new AtomicBoolean(false);
+    private static boolean libraryIsLoaded = false;
```

Comment:
I don't think this boolean is needed, at least not if you look at the JavaDoc for Java 7: http://docs.oracle.com/javase/7/docs/api/java/lang/Runtime.html#loadLibrary%28java.lang.String%29

The Android JavaDoc doesn't have the same clarity though, but it would be really odd behaviour if it loaded the same library twice?

http://developer.android.com/reference/java/lang/Runtime.html#loadLibrary(java.lang.String)